code/game/turfs/closed/walls.dm

```diff
@@ -33,6 +33,56 @@
 
 	var/list/dent_decals
 
+/turf/closed/wall/MouseDrop_T(mob/living/carbon/carbon_mob, mob/user)
+	..()
+	if(carbon_mob != user)
+		return
+	if(carbon_mob.is_leaning == TRUE)
+		return
+	if(carbon_mob.pulledby)
+		return
+	if(!carbon_mob.density)
+		return
+	carbon_mob.is_leaning = TRUE
+	var/turf/checked_turf = get_step(carbon_mob, turn(carbon_mob.dir, 180))
+	if(checked_turf == src)
+		carbon_mob.start_leaning(src)
+
+/mob/living/carbon/proc/start_leaning(obj/wall)
+	switch(dir)
+		if(SOUTH)
+			pixel_y += 11
+		if(NORTH)
+			pixel_y += -11
+		if(WEST)
+			pixel_x += 11
+		if(EAST)
+			pixel_x += -11
+	ADD_TRAIT(src, TRAIT_UNDENSE, LEANING_TRAIT)
+	ADD_TRAIT(src, TRAIT_EXPANDED_FOV, LEANING_TRAIT)
+	visible_message(span_notice("[src] leans against \the [wall]!"), \
+						span_notice("You lean against \the [wall]!"))
+	RegisterSignal(src, COMSIG_MOB_CLIENT_PRE_MOVE, PROC_REF(stop_leaning))
+	RegisterSignal(src, COMSIG_HUMAN_DISARM_HIT, PROC_REF(stop_leaning))
+	RegisterSignal(src, COMSIG_LIVING_GET_PULLED, PROC_REF(stop_leaning))
+	RegisterSignal(src, COMSIG_MOVABLE_TELEPORTING, PROC_REF(stop_leaning))
+	RegisterSignal(src, COMSIG_ATOM_DIR_CHANGE, PROC_REF(stop_leaning))
+	update_fov()
+
+/mob/living/carbon/proc/stop_leaning()
+	SIGNAL_HANDLER
+	UnregisterSignal(src, COMSIG_MOB_CLIENT_PRE_MOVE)
+	UnregisterSignal(src, COMSIG_HUMAN_DISARM_HIT)
+	UnregisterSignal(src, COMSIG_LIVING_GET_PULLED)
+	UnregisterSignal(src, COMSIG_MOVABLE_TELEPORTING)
+	UnregisterSignal(src, COMSIG_ATOM_DIR_CHANGE)
```

Comment:
You can do these all at once:
`UnregisterSignal(src, list(COMSIG_MOB_CLIENT_PRE_MOVE, COMSIG_HUMAN_DISARM_HIT, etc))`